### PR TITLE
Misra

### PR DESCRIPTION
**Pull request by guoch2**
Some changes for meeting the MISRA requirement.

### CODE REVIEW

File: hypervisor/arch/x86/guest/vlapic.c [hunk 1]
```diff
@@ -111,17 +111,21 @@ static uint16_t vm_apicid2vcpu_id(struct acrn_vm *vm, uint8_t lapicid)
 {
 	uint16_t i;
 	struct acrn_vcpu *vcpu;
+	uint16_t cpuid = 0xFFFFU;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vlapic.c]: You didn't address my comments in mail

File: hypervisor/arch/x86/guest/vlapic.c [hunk 2]
```diff
@@ -111,17 +111,21 @@ static uint16_t vm_apicid2vcpu_id(struct acrn_vm *vm, uint8_t lapicid)
...
 
 	foreach_vcpu(i, vm, vcpu) {
 		const struct acrn_vlapic *vlapic = vcpu_vlapic(vcpu);
 		if (vlapic_get_apicid(vlapic) == lapicid) {
-			return vcpu->vcpu_id;
+			cpuid = vcpu->vcpu_id;
 		}
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vlapic.c]: WHy miss break?

File: hypervisor/arch/x86/guest/vlapic.c [hunk 3]
```diff
@@ -659,6 +667,7 @@ vlapic_lvt_write_handler(struct acrn_vlapic *vlapic, uint32_t offset)
 {
 	uint32_t *lvtptr, mask, val, idx;
 	struct lapic_regs *lapic;
+	int error = 0;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vlapic.c]: comment in mail

**Comment by HuiHuangShi** [comment]
2149<br>22 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:get_vcpu_paging_mode&nbsp;&nbsp;offset:59:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:No brackets to then/else.<br><br>2.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:get_vcpu_paging_mode&nbsp;&nbsp;offset:61:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:No brackets to then/else.<br><br>3.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:get_vcpu_paging_mode&nbsp;&nbsp;offset:63:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:No brackets to then/else.<br><br>4.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:local_gva2gpa_common&nbsp;&nbsp;offset:76:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure contains UR data flow anomalies. : i<br><br>5.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:local_gva2gpa_common&nbsp;&nbsp;offset:81:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure contains UR data flow anomalies. : addr<br><br>6.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:local_gva2gpa_common&nbsp;&nbsp;offset:166:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Expression is not Boolean.<br><br>7.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:local_gva2gpa_common&nbsp;&nbsp;offset:196:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Expression is not Boolean.<br><br>8.filename:/hypervisor/arch/x86/guest/guest.c&nbsp;&nbsp;function:local_gva2gpa_pae&nbsp;&nbsp;offset:232:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Expression is not Boolean.<br><br>9.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vm_apicid2vcpu_id&nbsp;&nbsp;offset:114:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure name reused.                       : cpuid<br><br>10.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vm_apicid2vcpu_id&nbsp;&nbsp;offset:114:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Identifier reuse: procedure vs var. : cpuid<br><br>11.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vm_apicid2vcpu_id&nbsp;&nbsp;offset:119:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Signed/unsigned conversion without cast. :  (unsigned short and int): cpuid = vcpu -> vcpu_id<br><br>12.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vm_apicid2vcpu_id&nbsp;&nbsp;offset:119:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Use of mixed mode arithmetic. :  (unsigned short and int): cpuid = vcpu -> vcpu_id<br><br>13.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_get_lvtptr&nbsp;&nbsp;offset:649:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Declaration does not specify an array.       : lvt<br><br>14.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_get_lvtptr&nbsp;&nbsp;offset:652:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer assignment to wider scope.<br><br>15.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_lvt_write_handler&nbsp;&nbsp;offset:670:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Basic type declaration used. : int error<br><br>16.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_trigger_lvt&nbsp;&nbsp;offset:947:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure contains UR data flow anomalies. : lvt<br><br>17.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_trigger_lvt&nbsp;&nbsp;offset:999:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Use of mixed mode arithmetic. :  (unsigned int and signed char): ret = - 22<br><br>18.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_trigger_lvt&nbsp;&nbsp;offset:1003:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Signed/unsigned conversion without cast. :  (unsigned int and int): ret<br><br>19.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_trigger_lvt&nbsp;&nbsp;offset:1003:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Literal value requires a U suffix.           : 0<br><br>20.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_trigger_lvt&nbsp;&nbsp;offset:1011:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return type inconsistent. :  (int and unsigned int)<br><br>21.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_process_init_sipi&nbsp;&nbsp;offset:1205:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Type conversion without cast. :  (enum vm_state and enum vcpu_state): target_vcpu -> state == VCPU_INIT<br><br>22.filename:/hypervisor/arch/x86/guest/vlapic.c&nbsp;&nbsp;function:vlapic_icrlo_write_handler&nbsp;&nbsp;offset:1228:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Cyclomatic complexity greater than 20.<br><br>

File: hypervisor/arch/x86/guest/vlapic.c (hunk 2 above)
Comment (guoch2) [reply hypervisor/arch/x86/guest/vlapic.c]: sorry, I add it right now

**Comment by guoch2** [comment]
Should I fix all MISRA in the comment before requesting pull ?  Is there any way that I can run MISRA scanning locally or somewhere ?

File: hypervisor/arch/x86/guest/vlapic.c (hunk 3 above)
Comment (guoch2) [reply hypervisor/arch/x86/guest/vlapic.c]: ok, I changed it to uint32_t now.

**Comment by guoch2** [comment]
Unless fixing the two issues which Eddie pointed out,  I add two more patches to fix MISAR issue:
Here is my branch:   https://github.com/guoch2/acrn-hypervisor/tree/misra2

Is there any way that I could run  MISRA scanning before I send it the V2 patches out ?

**Comment by HuiHuangShi** [comment]
> Unless fixing the two issues which Eddie pointed out, I add two more patches to fix MISAR issue:
> Here is my branch: https://github.com/guoch2/acrn-hypervisor/tree/misra2
> 
> Is there any way that I could run MISRA scanning before I send it the V2 patches out ?

U can use pr scan,the feature that scan before pr is not finished.

File: hypervisor/arch/x86/guest/vlapic.c (hunk 1 above)
Comment (guoch2) [reply hypervisor/arch/x86/guest/vlapic.c]: Strange,  You can take a look at the source code.   I did change it .
Line 111:
https://github.com/guoch2/acrn-hypervisor/blob/misra/hypervisor/arch/x86/guest/vlapic.c.

**Pull request closed without merging**